ush/python/pygfs/task/atm_analysis.py

```diff
@@ -73,46 +70,42 @@ def __init__(self, config: Dict[str, Any], yaml_name: Optional[str] = None):
         # Extend task_config with local_dict
         self.task_config = AttrDict(**self.task_config, **local_dict)
 
-        # Create JEDI object
-        self.jedi = Jedi(self.task_config, yaml_name)
-
-    @logit(logger)
-    def initialize_jedi(self):
-        """Initialize JEDI application
-
-        This method will initialize a JEDI application used in the global atm analysis.
-        This includes:
-        - generating and saving JEDI YAML config
-        - linking the JEDI executable
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        ----------
-        None
-        """
+        # Create dictionary of JEDI objects
+        self.jedi = AttrDict()
 
-        # get JEDI-to-FV3 increment converter config and save to YAML file
-        logger.info(f"Generating JEDI YAML config: {self.jedi.yaml}")
-        self.jedi.set_config(self.task_config)
-        logger.debug(f"JEDI config:\n{pformat(self.jedi.config)}")
-
-        # save JEDI config to YAML file
-        logger.debug(f"Writing JEDI YAML config to: {self.jedi.yaml}")
-        save_as_yaml(self.jedi.config, self.jedi.yaml)
+        # atmanlvar
+        self.jedi['atmanlvar'] = Jedi(AttrDict(
```

Comment:
I think this whole thing is a sign this class is trying to do too many independent things. The class should only be instantiated for increment or for variational (one executable, one algo yaml, etc.), not try to cover both as is being done here. If the differences are just in these settings (as seems to be the case), this can be done without a new class by just passing in the relevant values. If there is coding that needs to be different, it can with child classes that inherit from `AtmAnalysis`.

Right now, I don't think the two are even created in the same job, so this could be as easy as using the same envvar to specify the executable and the algo yaml in the separate config files for each job instead of separate names for each type. i.e. call them `JEDIEXE` and `JCB_ALGO_YAML`, and just specify them differently in the config file. Since the configs are only sourced in their respective jobs, there would be no name collision (and each could still use a different `@_@` variable, if you )

This should allow you to revert a lot of these changes that are adding messiness to the method calls and keep things nice and straightforward.

Reply:
@WalterKolczynski-NOAA What you're seeing is indeed one Jedi object per JEDI executable. AtmAnalysis.jedi, for example, is a dictionary of Jedi objects, each key pointing to one object for one executable. This structure, along with initializing each JEDI object with a dictionary of required parameters (executable path, JCB algo, etc), will allow us to put all these parameters in one big, clean YAML file to configure all the JEDI executable in a set of jobs like the atm or marine analysis.

The reason for getting rid of JEDIEXE as a generic environment variable and replacing with one JEDIEXE_*, is this allows us to initialize all JEDI executables in the init job. The way things were before in AtmEnsAnalysis, for example, we would initialize the atmensanlobs executable in the atmensanlinit job, but the rest of the JEDI executable would have to be initialized in their respective run jobs. For example, the atmensanlfv3inc executable had to be initialized in the atmensanlfv3inc job where that executable is actually run. I think we'd like to keep all initialization in the atmensanlinit job. Having one JEDIEXE and one JCB_ALGO per config file prevents us from doing this.